test/unit/legacy-data.html

```diff
@@ -0,0 +1,278 @@
+<!doctype html>
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<html>
+<head>
+  <meta charset="utf-8">
+  <script src="../../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../../web-component-tester/browser.js"></script>
+  <link rel="import" href="../../polymer.html">
+  <link rel="import" href="../../lib/legacy/legacy-data-mixin.html">
+</head>
+<body>
+
+  <script>
+  </script>
+  
+  <dom-module id="x-data">
+    <script>
+      HTMLImports.whenReady(() => {
+        Polymer({
+          is: 'x-data',
+          observers: [
+            'static("static")',
+            'one(a)',
+            'two(b, c)',
+            'throws(d)'
+          ],
+          created() {
+            this.one = sinon.spy();
+            this.two = sinon.spy();
+            this.static = sinon.spy();
+          },
+          throws() {
+            throw new Error('real error');
+          }
+        });
+      });
+    </script>
+  </dom-module>
+
+  <test-fixture id="declarative-one">
+    <x-data a="a"></x-data>
+  </test-fixture>
+
+  <test-fixture id="declarative-two">
+    <x-data a="a" b="b"></x-data>
+  </test-fixture>
+
+  <test-fixture id="declarative-all">
+    <x-data a="a" b="b"></x-data>
+  </test-fixture>
+
+  <script>
+  suite('imperative', () => {
```

Comment:
This suite of tests has a lot of duplication I think we could do with some test helpers to significantly clean up the test implementation to make the code easier to maintain and easier to reason about.